packages/react-tabster/src/TabsterProvider.tsx

```diff
@@ -1,26 +1,30 @@
 import * as React from 'react';
-import { makeMergePropsCompat, useMergedRefs } from '@fluentui/react-utilities';
+import { assign, ComponentState, makeMergeProps, useMergedRefs } from '@fluentui/react-utilities';
 import { getCurrentTabster, createTabster } from 'tabster';
 import { internal__TabsterContext, TabsterContextValue } from './TabsterContext';
 
 export interface TabsterProviderProps extends React.HTMLAttributes<HTMLElement> {
   dir?: 'rtl' | 'ltr';
-  document: Document | undefined;
+  document?: Document | undefined;
   /**
    * The root is automatically set as the `body` element of the ownerDocument.
    * This prop needs to be set if a custom root is used
    */
   customRoot?: boolean;
 }
 
-export interface TabsterProviderState extends TabsterProviderProps {
-  dir: 'ltr' | 'rtl';
-  document: Document | undefined;
-  contextValue: TabsterContextValue | undefined;
-}
+export type TabsterProviderState = ComponentState<
+  React.Ref<HTMLElement>,
+  TabsterProviderProps & {
+    contextValue: TabsterContextValue | undefined;
```

Comment:
```suggestion
    contextValue: TabsterContextValue | undefined;
    document: Document | undefined;
```

In `state` - `document` is not optional, it should be defined or be `undefined`.